third_party/nvfuser/csrc/lower_thread_predicate.cpp

```diff
@@ -264,6 +275,21 @@ void ThreadPredicateMap::updateBitSet(const Expr* expr) {
     const auto bcast_reset_mask = ~(this_input_preds & id_bcasts);
     this_input_preds &= bcast_reset_mask;
 
+    // If the input is on shared or global memory and is predicated,
+    // and the output is parallelized by the predicate type, a RAW
+    // sync is automatically inserted, so the predication can be
+    // cleared.
+    if (tv_inp->getMemoryType() == MemoryType::Shared ||
+        tv_inp->getMemoryType() == MemoryType::Global) {
+      auto preds_to_clear = (this_input_preds & output_ptypes);
+      // If it's on shared memory, only TID predicates can be
+      // cleared.
+      if (tv_inp->getMemoryType() == MemoryType::Shared) {
```

Comment:
Should the same thing in global memory not clear all threads?

Reply:
This disables clearing of the BID flags in the case of shared memory. Since with gmem both TID and BID flags can be cleared, nothing needs to be done. 

I realized this is double negative. I'll see if I could make it more clear.

Reply:
Ah I got it, no worries, you're fine.

Reply:
I changed a little anyway.